tests/hazmat/primitives/utils.py

```diff
@@ -528,7 +528,10 @@ def _check_rsa_private_numbers(skey):
     assert pkey
     assert pkey.e
     assert pkey.n
-    assert skey.d
+    assert skey.d in (
+        rsa.rsa_recover_private_exponent(pkey.e, skey.p, skey.q),
+        rsa._rsa_recover_euler_private_exponent(pkey.e, skey.p, skey.q),
+    )
```

Comment:
There aretwo different ways to calculate valid/working `d` given `e`, `p`, `q`.

In my opinion, it makes sense to expose _only_ the more modern one using the Carmichael totient function for public usage, given that it's slightly more computationally efficient to use the keys in this form, and that some standards require it:

https://en.wikipedia.org/wiki/RSA_(cryptosystem)#:~:text=In%20the%20original,smaller%20equivalent%20exponent.